### PR TITLE
added range_loagAUC as an evaluator

### PR DESCRIPTION
**Pull request by LanceKnight**
Author: Yunchao "Lance" Liu (lanceknight26@gmail.com)
Calculate logAUC in a certain FPR range (default range: [0.001, 0.1]).
This was used by previous methods [1] and the reason is that only a
small percentage of samples can be selected for experimental tests in
consideration of cost. This means only molecules with very high
predicted activity values can be worth testing, i.e., the decision
threshold is high. And the high decision threshold corresponds to the
left side of the ROC curve, i.e., those FPRs with small values. Also,
because the threshold cannot be predetermined, the area under the curve
is used to consolidate all possible thresholds within a certain FPR
range. Finally, the logarithm is used to bias smaller FPRs. The higher
the logAUC[0.001, 0.1], the better the performance.

A perfect classifer gets a logAUC[0.001, 0.1] ) of 1, while a random
classifer gets a logAUC[0.001, 0.1] ) of around 0.0215 (See [2])

References:
[1] Mysinger, M.M. and B.K. Shoichet, Rapid Context-Dependent Ligand
Desolvation in Molecular Docking. Journal of Chemical Information and
Modeling, 2010. 50(9): p. 1561-1573.
[2] Mendenhall, J. and J. Meiler, Improving quantitative
structure–activity relationship models using Artificial Neural Networks
trained with dropout. Journal of computer-aided molecular design,
2016. 30(2): p. 177-189.
:param true_y: numpy array of the ground truth. Values are either 0 (
inactive) or 1(active).
:param predicted_score: numpy array of the predicted score (The
score does not have to be between 0 and 1)
:param FPR_range: the range for calculating the logAUC formated in
(x, y) with x being the lower bound and y being the upper bound
:return: a numpy array of logAUC of size [1,1]

### CODE REVIEW

**Comment by kexinhuang12345** [comment]
looking great! is there a test case that you could help write up? something like this: https://tdcommons.ai/functions/data_evaluation/#precision-at-recall-of-k

**Comment by LanceKnight** [comment]
**Description**
In a realistic setting, only a small percentage of samples can be selected for experimental tests in consideration of cost. This means only molecules with very high predicted activity values can be worth testing, i.e., the decision threshold is high. And the high decision threshold corresponds to the left side of the ROC curve, i.e., those FPRs with small values. Also, because the threshold cannot be predetermined, the area under the curve is used to consolidate all possible thresholds within a certain FPR
range. Finally, the logarithm is used to bias smaller FPRs. See [1] for reference. 

Default range: [0.001, 0.1]. The higher the logAUC[0.001, 0.1], the better the performance.
A perfect classifer gets a logAUC[0.001, 0.1] ) of 1, while a random classifer gets a logAUC[0.001, 0.1] ) of around 0.0215 (See [2])

:param true_y: numpy array of the ground truth. Values are either 0 (
inactive) or 1(active).
:param predicted_score: numpy array of the predicted score (The
score does not have to be between 0 and 1)
:param FPR_range: the range for calculating the logAUC formated in
(x, y) with x being the lower bound and y being the upper bound
:return: a numpy array of logAUC of size [1,1]

References:
[1] Mysinger, M.M. and B.K. Shoichet, Rapid Context-Dependent Ligand
Desolvation in Molecular Docking. Journal of Chemical Information and
Modeling, 2010. 50(9): p. 1561-1573.
[2] Mendenhall, J. and J. Meiler, Improving quantitative
structure–activity relationship models using Artificial Neural Networks
trained with dropout. Journal of computer-aided molecular design,
2016. 30(2): p. 177-189.


```
from numpy import random
from tdc import Evaluator

# Prepare testing samples
zeros = np.zeros((99900,1))
ones =  np.ones((100,1))
true_y = np.concatenate((zeros, ones)) # true_y has 99900 actives and 100 inactives

# Prepare evaluator
evaluator = Evaluator(name = 'range_logAUC')

# Test case 1: random classifier
pred = random.random((100000,1))
score = evaluator(true_y, pred) # default false positive rate (FPR) ranges from 0.001 to 0.1. Change this range according to the active/inactive ratio.
# score ~= 0.0215. The more number of total samples, the more likely the score is close to 0.0215

# Test case 2: perfect classifier
pred = np.ones((100000,1))
scorec = calculate_logAUC(true_y, pred)
# score = 1
```

**Comment by kexinhuang12345** [comment]
Thank you Lance! We will test it and release it in the next version!

**Comment by LanceKnight** [comment]
Sounds good. Thank you! Let me know if anything if needed :)

**Comment by kexinhuang12345** [comment]
Just tested, and worked like a charm. Thanks Lance for the contribution!

**Review by kexinhuang12345 — APPROVED** [review]

**Comment by LanceKnight** [comment]
No problem. Glad that I can help :)
